qiskit_experiments/calibration/experiments/rabi.py

```diff
@@ -0,0 +1,337 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Rabi amplitude Experiment class."""
+
+from typing import Callable, List, Optional, Tuple, Union
+import numpy as np
+
+from qiskit import QiskitError, QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.qobj.utils import MeasLevel
+from qiskit.providers import Backend
+from qiskit.pulse import ScheduleBlock
+import qiskit.pulse as pulse
+from qiskit.providers.options import Options
+
+from qiskit_experiments.base_analysis import BaseAnalysis, ExperimentData, AnalysisResult
+from qiskit_experiments.base_experiment import BaseExperiment
+from qiskit_experiments.analysis.curve_fitting import curve_fit
+from qiskit_experiments.data_processing.processor_library import get_to_signal_processor
+from qiskit_experiments.analysis import plotting
+
+
+class RabiAnalysis(BaseAnalysis):
+    r"""Rabi analysis class based on a fit to a cosine function.
+
+    Analyse a Rabi experiment by fitting it to a cosine function
+
+    .. math::
+        a * \cos(b*x) + c
+
+    The y-values will be normalized to the range 0-1.
+    """
+
+    @classmethod
+    def _default_options(cls):
+        return Options(
+            amp_guess=0.5,
+            freq_guesses=np.linspace(0, 5 * np.pi, 10),
+            offset_guess=0.5,
+            phase_guess=None,
+            amp_bounds=(-1, 1),
+            freq_bounds=(0, np.inf),
+            offset_bounds=(0, 1),
+            phase_bounds=(-np.pi, np.pi),
+        )
```

Comment:
Maybe could share some code with T2*. Either way, we've talked about standardization of options, and this is particularly visible here, when comparing with user p0 and bounds that's in T2*.

Reply:
@merav-aharoni

Reply:
See also curve analysis PR. This will be standardized with p0 and bounds. These values are returned as a dict with key matched with variables in series definition.